src/metabase/xrays/automagic_dashboards/combination.clj

```diff
@@ -78,13 +79,16 @@
    {{:keys [database]} :root :keys [source query-filter]}]
   (let [source-table (if (->> source (mi/instance-of? :model/Table))
                        (-> source u/the-id)
-                       (->> source u/the-id (str "card__")))]
+                       (->> source u/the-id (str "card__")))
+        model?       (and (mi/instance-of? :model/Card source)
+                          (card/model? source))]
     (assoc ground-metric-with-dimensions
            :dataset_query {:database database
                            :type     :query
                            :query    (cond-> (assoc metric-definition
                                                     :source-table source-table)
-                                       query-filter (assoc :filter query-filter))})))
+                                       (and (not model?)
```

Comment:
Do we not also need to check whether it has an aggregation?

Reply:
Hmmm... I think the question here is - if it's a model, all the filtering has already happened inside it, so we don't add it on top. If it's a direct table - then yeah, we should carry the filtering defined in a card.